tests/unit/lms/models/lti_user_test.py

```diff
@@ -31,3 +33,22 @@ def test_is_learner(self, roles, is_learner):
         lti_user = factories.LTIUser(roles=roles)
 
         assert lti_user.is_learner == is_learner
+
+    def test_from_names(self, display_name):
+        # Use factories.LTIUser to randomly generate a dict of all the kwargs
+        # that you would need to pass to LTIUser(**kwargs).
+        kwargs = factory.build(dict, FACTORY_CLASS=factories.LTIUser)
+        # Delete the display_name kwarg because from_names() doesn't use it.
+        del kwargs["display_name"]
+
+        user = LTIUser.from_names(
+            given_name="Gareth", family_name="Smith", full_name="", **kwargs,
+        )
+
+        display_name.assert_called_once_with("Gareth", "Smith", "")
+        assert user.display_name == display_name.return_value
+
+
+@pytest.fixture(autouse=True)
+def display_name(patch):
+    return patch("lms.models.lti_user.display_name")
```

Comment:
`LTIUser` and `display_name()` are a pretty small-scale case of separate units, but I still decided to patch `display_name()` in `LTIUser`'s tests because:

* `display_name()` has its own separate unit tests and those should be responsible for `display_name()`'s test coverate. `LTIUser`'s unit tests aren't responsible for testing `display_name()`. We don't want `LTIUser`'s unit tests contributing to `display_name()`'s test coverage, and potentially hiding coverage holes in `display_name()`'s actual unit tests

* `display_name()` has its own separate unit tests and those are responsible for failing or crashing if there's a bug in `display_name()`. We don't want `LTIUser`'s unit tests failing or crashing because of bugs in `display_name()`. They should only fail or crash because of bugs in `LTIUser`

* It makes the tests read more clearly about what `LTIUser`'s responsibilities actually are. `assert user.display_name == display_name.return_value` tells you that `LTIUser`'s job is just to blindly use whatever `display_name()` returns as the display name. Something like `assert user.display_name == "Gareth Smith"`, which depends on the particular value returned by 'display_name()`'s actual implementation, confuses what the `LTIUser` tests are about

Reply:
I think this is the downside of having the factory function. The length cost isn't the main code, but the tests. By creating this interface between `LTIUser` and `display_name()` and coupling them, we now have to test and police it.

If it was simply the users choice to use `display_name()` and provide the string to `LTIUser()` init method, we wouldn't have to.

I believe (correct me if wrong), this code will only ever be called once, at which point we will presumably have to test _that_ boundary too, mocking `from_names`. If that caller did both we could kill two birds with one stone there.